File: backend-node/src/utils/logger.js
/**
 * Request logging middleware
 */
const requestLogger = (req, res, next) => {
  const timestamp = new Date().toISOString();
  console.log(`[${timestamp}] ${req.method} ${req.path}`);
  next();
};

/**
 * Log info messages
 */
const logInfo = (message, meta = {}) => {
  console.log(`[INFO] ${message}`, meta);
};

/**
 * Log error messages
 */
const logError = (message, error = {}) => {
  console.error(`[ERROR] ${message}`, error);
};

/**
 * Log warning messages
 */
const logWarning = (message, meta = {}) => {
  console.warn(`[WARNING] ${message}`, meta);
};

module.exports = {
  requestLogger,
  logInfo,
  logError,
  logWarning
};

File: ml-service/app/utils/logger.py
"""
Utility functions for ML service
"""
from datetime import datetime


def log_info(message: str, data: dict = None):
    """Log info message"""
    timestamp = datetime.now().isoformat()
    print(f"[INFO] [{timestamp}] {message}")
    if data:
        print(f"  Data: {data}")


def log_error(message: str, error: Exception = None):
    """Log error message"""
    timestamp = datetime.now().isoformat()
    print(f"[ERROR] [{timestamp}] {message}")
    if error:
        print(f"  Error: {str(error)}")


def log_warning(message: str, data: dict = None):
    """Log warning message"""
    timestamp = datetime.now().isoformat()
    print(f"[WARNING] [{timestamp}] {message}")
    if data:
        print(f"  Data: {data}")
